### PR TITLE
setup.sh improvements

### PR DESCRIPTION
**Pull request by KellerFuchs**
- Uses the cleaner way of handling the package set.
- Sets up `/etc` with correct file permissions.
- Do not setup SELinux (it was removed in #66).
- Correctly handles `grub`.
- Sets access rights that are consistent with what is currently deployed.

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -33,19 +32,13 @@ git init
 git remote add origin https://github.com/hashbang/shell-etc.git
 git fetch --all
 git reset --hard origin/master
-git clean -f
-chmod 0600 /etc/sssd/sssd.conf
-
-# Disable users knowing about other users
-chmod o-r /var/run/utmp # Used by who(1)
```
Comment (daurnimator) [inline setup.sh]: did you mean to remove the utmp and wtemp hiding?

File: setup.sh [hunk 2]
```diff
@@ -33,19 +32,13 @@ git init
...
-chmod o-r /var/log/wtmp # Used by last(1)
-chmod o-r /var/log/lastlog # Used by lastlog(8)
-chmod o-r /home # prevent listing of all home dirs.
```
Comment (daurnimator) [inline setup.sh]: why was this removed?

**Comment by daurnimator** [comment]
Except for 2552d79e8f1ad4cc88c99fda7f487b42347a0dcc looks good to me.

File: setup.sh [hunk 3]
```diff
@@ -33,8 +32,8 @@ git init
 git remote add origin https://github.com/hashbang/shell-etc.git
 git fetch --all
 git reset --hard origin/master
```
Comment (daurnimator) [inline setup.sh]: should set tracking branch too.
`git branch --set-upstream-to=origin/master master`

File: setup.sh [hunk 4]
```diff
@@ -33,8 +32,8 @@ git init
 git remote add origin https://github.com/hashbang/shell-etc.git
 git fetch --all
```
Comment (daurnimator) [inline setup.sh]: Would prefer `git fetch origin master` here: no need to pull in all the branches on the server.

File: setup.sh (hunk 1 above)
Comment (KellerFuchs) [inline setup.sh]: Yes.  As said in the commit comment, they are not currently hidden on the servers.

File: setup.sh (hunk 4 above)
Comment (KellerFuchs) [inline setup.sh]: Good point

File: setup.sh (hunk 1 above)
Comment (daurnimator) [inline setup.sh]: yeah. we should fix that :)
Could you remove commit 2552d79e8f1ad4cc88c99fda7f487b42347a0dcc from this PR for discussion elsewhere?

**Comment by daurnimator** [comment]
bootstrapping question: how do we actually run setup.sh considering that shell-etc is not pulled in yet?

This matters, as if pulling in shell-etc adds in new debian sources, we'll possibly need to run the apt update && apt upgrade flow multiple times.

**Comment by KellerFuchs** [comment]
@daurnimator Basically, you either scp it to the remote box or fetch it from GitHub.

**Comment by ChickenNuggers** [comment]
> Except for 2552d79 looks good to me.

Currently 2552d79 mirrors shell servers; perhaps we should either:
 * change etckeeper permissions to mirror this, or
 * keep 2552d79

**Comment by KellerFuchs** [comment]
@ChickenNuggers The permissions in questions are not managed by `etckeeper`, FWIW.

**Comment by ChickenNuggers** [comment]
Oh; after re-reading, it's in /var apparently. My bad. Either way, I do remember us attempting to do this on the servers and I also remember it being reverted. @lrvick was there a reason behind this, if you can recall?

**Comment by KellerFuchs** [comment]
@daurnimator Backed out the `chmod utmp` commit.  OK for merging this part?

**Comment by daurnimator** [comment]
Yep :)
